### ISSUE TITLE
[Feature request] Don't show/receive message from blocked user in group

### ISSUE THREAD

**Issue by CzBiX** [telegramdesktop/tdesktop#1025]

**Comment by auchri** [comment]
Do you see messages from blocked users in the iOS- or Android-app?

**Comment by CzBiX** [comment]
@auchri Yes, I can see that in the iOS/Android app.

**Comment by auchri** [comment]
So it's not really a issue of telegram desktop :)

The messages should be filtered by the server

**Comment by foxmajik** [comment]
But messages from blocked users are not blocked in chats on the desktop client when that user is blocked using the desktop client.
If I block someone I still see messages in the chat from them on the desktop client.

See the attached screen shots:

<img width="536" alt="screenshot 2015-11-08 15 56 39" src="https://cloud.githubusercontent.com/assets/300165/11023560/5932a030-8631-11e5-808c-0fa598db2a4b.png">
<img width="633" alt="screenshot 2015-11-08 15 56 27" src="https://cloud.githubusercontent.com/assets/300165/11023561/593341d4-8631-11e5-81eb-983af64b5198.png">

**Comment by BalooUriza** [comment]
I also agree that this is a badly needed feature.

**Comment by MarcusWolschon** [comment]
Definately needed. Back in IRC days blocking people in group chats was ist perfectly normalerweise and the resulting interaction worked for ages.

**Comment by urbanhusky** [comment]
_But not seeing what they have written breaks context in discussions and replies!_
That never really was a problem in IRC and I'd argue that this is more of a constructed issue.
Plus, we've got more UI/UX options now: instead of completely hiding their presence, hide their text and replace it with a de-emphasised placeholder text like "user is blocked" - maybe even allow a "show hidden text" to check what they have written - or something like that.

But please, don't leave me with the options of either leaving the group completely (not acceptable) or still having to deal with that user.

**Comment by stek29** [comment]
@MarcusWolschon server side issue tracker is not public.
API issue label is also used for issues which can't be implemented in tdesktop only because of platform consistency.

**Comment by urbanhusky** [comment]
If it cannot be implemented in desktop for consistency, why not implement it on _all_ platforms? 
The hidden/de-emphasised approach should not need server/API support.

**Comment by MarcusWolschon** [comment]
Looks like Telegre.at forked the code and implemented it on Android, Desktop (no iOS yet) and Web.

https://telegre.at/#download

**Comment by chanakyabtc** [comment]
> Looks like Telegre.at forked the code and implemented it on Android, Desktop (no iOS yet) and Web.
> 
> https://telegre.at/#download

Thank you so much dude. This is awesome. Lifesaver. Got an awesome group with just two cancerous triggering guys lol this saved the day

**Comment by 411an** [comment]
> https://telegre.at/#download

This application actually dont help to ignore blocked people. It just turns color of their messages in grey.

Besides the ignored person still can send pictures in the group chat, and you can see them.

I writed to developer of that fork, he said he was not going to make the full ignoring with hiding messages from blocked user.

Does anybody know other forks with real ignore of blocked users?

**Comment by RememberTheAir** [comment]
![image](https://user-images.githubusercontent.com/16143208/61454791-24c09200-a962-11e9-9174-ee3088b1145d.png)

[Twitter thread](https://twitter.com/buunicubus/status/1151604995487338496)

**Comment by SmartManoj** [comment]
> Looks like Telegre.at forked the code and implemented it on Android, Desktop (no iOS yet) and Web.
> 
> https://telegre.at/#download

Android and web link ?

**Comment by gmolledaj** [comment]
Why is it closed if there are people who do not want Telegram to avoid putting up messages from certain contacts in family groups, themes, etc.?

**Comment by Aokromes** [comment]
because https://i.imgur.com/E6OhuUG.png

**Comment by gmolledaj** [comment]
But it does not have to do with an API, the application itself has access to the name of the contacts and can delete the contact's message on the screen. Or is it not possible to analyze the internal code of the messages as if it were HTML from a website before putting them on the screen? Thus, the text to be shown could be modified without the messages of the blocked users.

**Comment by Aokromes** [comment]
> But it does not have to do with an API, the application itself has access to the name of the contacts and can delete the contact's message on the screen. Or is it not possible to analyze the internal code of the messages as if it were HTML from a website before putting them on the screen? Thus, the text to be shown could be modified without the messages of the blocked users.

it has to do, it's needed to block it on all clients, not on telegram desktop only.

**Comment by 411an** [comment]
OK I did it for all suffering people

https://sites.google.com/view/ignoram/

**Comment by gmolledaj** [comment]
No. The Telegram system is different from WhatsApp, in Telegram the messages are on the server, a single list of messages that all members of the group see. You cannot hide or delete anything in the group because it would be deleted for all members of the group. It must be something to implement in the client application, each client will hide, not delete, the blocked message. It must be programmed in each telegram application separately: desktop, android app, ios app, telegram-web.

**Comment by foxmajik** [comment]
>  It must be something to implement in the client application, each client will hide, not delete, the blocked message. It must be programmed in each telegram application separately: desktop, android app, ios app, telegram-web.

Yes. We know that. That's what we've been asking for literally since Telegram became a thing. You don't need to delete the message from the server. Just don't f------ show it to me. I don't want to see it.

**Comment by trochdewei** [comment]
This is the feature that we really need, hope it will be added at some point

**Comment by aphirst** [comment]
@411an Are you able to host a forked repo or some .patch files for this? It'd be very useful for Linux users to have something like that on which to base a user package.

**Comment by 411an** [comment]
> 
> 
> @411an Are you able to host a forked repo or some .patch files for this? It'd be very useful for Linux users to have something like that on which to base a user package.

The first problem is the telegram code changes every time when I make new version. They change the code, add strings, conditions or even new functions. When I created site for Ignoram - I made the page with code and path instructions. But it just didnt work for next version. I changed it - but for next version it didnt work again. So now I offer for download application only. But I will think about udating the page every time when I updating application.

The second problem - I dont know linux :(

It would be really good if somebody who better than me in programming do this. But most of them dont understand whats wrong with telegram, so I trying to do what I can.
